javascript/pkce_oauth/README.md

```diff
@@ -122,24 +130,39 @@ Now throw all this code into your '/first-step-auth' endpoint, and lets talk abo
 app.get('/first-step-auth', (req, res) => {
     console.log('In first-step-auth');
     
-    let result = {};
+    const state = getRandomString();
+    const codeVerifier = getRandomString();
+
+    // We use crypto to generate sha256 bytes from the codeVerifier, and then digest the bytes with base64
+    let codeChallenge = crypto.createHash('sha256')
+        .update(codeVerifier)
+        .digest('base64');
 
-    let state = getRandomToken();
-    let codeVerifier = getRandomToken();
-    let codeChallenge = sha256(codeVerifier);
+    // Because the client will be sending the challenge in the url, lets url encode it for them
```

Comment:
So, there's a difference between "URL encoding a (base 64 encoded string)" and "(base 64 URL encoding) a string." The first performs `%` escaping of special characters, and the second just uses characters that don't need escaping. This step on line 142 isn't doing any encoding, it's just replacing/"escaping" the characters, so this comment might be confusing.

Reply:
Ah. Good catch. Do you have a suggestion for a better way to say this? Perhaps:

// Currently we have a base64 encoded string. PKCE calls for a base64url encoded string, so lets convert it.

Reply:
Yeah, maybe something like "The `crypto` library does not support base 64 URL encoding, so we must convert the result to be compliant" so that we still motivate why it's done in two steps.